bundles/ui/org.eclipse.smarthome.ui.icon/src/main/java/org/eclipse/smarthome/ui/icon/internal/IconServlet.java

```diff
@@ -97,21 +102,62 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
             return;
         }
 
-        String filename = StringUtils.substringAfterLast(req.getRequestURI(), "/");
-
-        String iconName = StringUtils.substringBeforeLast(filename, ".");
+        String category = getCategory(req);
+        Format format = getFormat(req);
```

Comment:
Yes, this is my assumption. If the icon set meta-data claims that it supports SVG, this should be true for all icons (even the custom ones).